### PR TITLE
Prevent UPE onboarding checkboxes getting unchecked 

### LINKED ISSUE #5193: Payment methods getting unchecked in UPE Onboarding form
### Describe the bug

The issue is observed when you first enable UPE ( Additional Payment Methods ) , or disable it, and re-enable it. On the `Add Additional Payments` wizard, that appears, if you click multiple payment methods in quick succession ( within 1.5 seconds ) , few of them get automatically unchecked at the end of about 1.5 seconds. 

### To Reproduce

1. Make sure UPE is disabled, within `Payments > Settings` you should see the enable option. Please click that 

image

If UPE is already enabled, click the three dots to the top corner of the section , and disable it. and then enable to replicate the error

2. Click on `Enable in your store`. It should open the `Add additional payment methods` wizard. 
3. Try selecting a few payment methods there very quickly. 
4. You will see some of them getting automatically unchecked

### Actual behavior
Payment methods get automatically unchecked. 

### Screenshots

upe-onboarding-Screen Capture on 2022-11-25 at 13-58-35

### Expected behavior
The methods should remain checked or unchecked, depending on actual selection by user. 

### Desktop (please complete the following information):

* OS: macOS Version 13.0.1 (Build 22A400)
* Browser : Chrome
* Version 107.0.5304.110 

### Additional context
Issue is very similar to #4629, but on a different component.

### PR DESCRIPTION
**Pull request by nagpai**
Fixes #5193 

##### Summary of the issue
On the UPE onboarding form, thats seen when we first enable UPE ( additional payment options ), there is a checkbox selector in the onboarding wizard. Selecting multiple options within a short span of 1.5 seconds, made only the first box selected at the end of it, and subsequently selected boxes, unchecked. 

**Issue ( Before ) :** 

upe-onboarding-Screen Capture on 2022-11-25 at 13-58-35

The issue was happening due to the complete `paymentMethodsState` being sent to the state updation hook which [expects two parameters](https://github.com/Automattic/woocommerce-payments/blob/ed3976f66ab2433c7184c629135f115209c7a782/client/additional-methods-setup/upe-preview-methods-selector/add-payment-methods-task.js#L40) - the changed payment method name, and its status (boolean) 

The PR corrects the call to the change handler and sends the specific parameters, instead of the complete updated state. 

**Result (after):** 
after

#### Testing instructions

1. Make sure UPE is disabled. Within `Payments > Settings` you should see the enable option. Please click that 

image

If UPE is already enabled, click the three dots to the top corner of the section , and disable it. and then enable to replicate the error.

2. Click on `Enable in your store`. It should open the `Add additional payment methods` wizard. 
3. Try selecting a few payment methods there very quickly. 
4. You should now see all of them remaining checked. None of them should get unchecked automatically
5. Uncheck one or two of them and click on `Add payment methods` and complete the wizard
6. On the settings page that loads, you should see the selection of UPE methods matching the final selection on the onboarding wizard earlier.

The PR passes all existing tests that seem adequate to me. Do guide me if anything more is needed. 

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply) - N. A. 

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Review by frosso — APPROVED** [review]
Clean and concise, looks good! 👍

File: client/additional-methods-setup/upe-preview-methods-selector/add-payment-methods-task.js [hunk 1]
```diff
@@ -158,8 +158,7 @@ const AddPaymentMethodsTask = () => {
 	);
 
 	const completeActivation = ( itemId ) => {
-		paymentMethodsState[ itemId ] = true;
-		handlePaymentMethodChange( paymentMethodsState );
+		handlePaymentMethodChange( itemId, true );
```
Comment (kalessil) [inline client/additional-methods-setup/upe-preview-methods-selector/add-payment-methods-task.js]: Would it be possible to rename `itemId` (to e.g. `method`) - by the context, it seems to be a payment method in there (but I'm not 100% sure)?

File: client/additional-methods-setup/upe-preview-methods-selector/add-payment-methods-task.js [hunk 2]
```diff
@@ -193,8 +192,7 @@ const AddPaymentMethodsTask = () => {
 				completeActivation( itemId );
 			}
 		} else {
-			paymentMethodsState[ itemId ] = false;
-			handlePaymentMethodChange( paymentMethodsState );
+			handlePaymentMethodChange( itemId, false );
```
Comment (kalessil) [inline client/additional-methods-setup/upe-preview-methods-selector/add-payment-methods-task.js]: Same here.

**Review by kalessil — CHANGES_REQUESTED** [review]

File: client/additional-methods-setup/upe-preview-methods-selector/add-payment-methods-task.js (hunk 1 above)
Comment (nagpai) [reply client/additional-methods-setup/upe-preview-methods-selector/add-payment-methods-task.js]: Thanks for calling this out, I have renamed it to `method`  - 55ec5d7

File: client/additional-methods-setup/upe-preview-methods-selector/add-payment-methods-task.js (hunk 2 above)
Comment (nagpai) [reply client/additional-methods-setup/upe-preview-methods-selector/add-payment-methods-task.js]: Thanks, I have updated it here - 55ec5d7

**Review by kalessil — APPROVED** [review]
LGTM!

**Pull request merged**